static/js/components/CouponNotificationDialog.js

```diff
@@ -0,0 +1,102 @@
+import type { Course } from '../flow/programTypes';
+
+type CouponNotification = {
+  coupon:               Coupon,
+  couponProgram:        ?AvailableProgram,
+  couponCourse:         ?Course,
+  open:                 boolean,
+  setDialogVisibility:  (v: boolean) => void,
+};
+
+const CouponNotificationDialog = (
+  { coupon, couponProgram, couponCourse, open, setDialogVisibility }: CouponNotification
+) => {
+  const {
+    amount,
+    amount_type: amountType,
+    content_type: contentType,
+    program_id: programId,
+    object_id: objectId,
+  } = coupon;
+  let programName;
+  if ( couponProgram ) {
+    programName = `the ${couponProgram.title} MicroMasters program`;
+  } else {
+    programName = `program ID ${programId}`;
+  }
+
+  let title, message;
+  if ( contentType === COUPON_CONTENT_TYPE_PROGRAM ) {
+    if ( amountType === COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT ) {
+      title = `Coupon applied: ${amount.times(100)}% off each course!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of { amount.times(100).toString() }% off</strong> the price
+        of <strong>each</strong> course in { programName }.
+      </p>;
+    } else if ( amountType === COUPON_AMOUNT_TYPE_FIXED_DISCOUNT ) {
+      title = `Coupon applied: $${amount} off each course!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of ${ amount.toString() } off</strong> the price
+        of <strong>each</strong> course in { programName }.
+      </p>;
+    }
+  } else {
+    let courseName;
+    if ( couponCourse ) {
+      courseName = `${couponCourse.title} in ${programName}`;
+    } else {
+      courseName = `course ID ${objectId} in ${programName}`;
+    }
+    if ( amountType === COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT ) {
+      title = `Coupon applied: ${amount.times(100)}% off!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of { amount.times(100).toString() }% off</strong> the price
+        of { courseName }.
+      </p>;
+    } else if ( amountType === COUPON_AMOUNT_TYPE_FIXED_DISCOUNT ) {
+      title = `Coupon applied: $${amount} off!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of ${ amount.toString() } off</strong> the price
+        of { courseName }.
+      </p>;
+    }
+  }
+
+  const okButton = <Button
+    type='ok'
+    key='ok'
+    className="primary-button ok-button"
+    onClick={() => setDialogVisibility(false)}>
+    OK
+  </Button>;
+
+  return <Dialog
+    title={title}
+    titleClassName="dialog-title"
+    contentClassName="dialog coupon-notification-dialog"
+    className="coupon-notification-dialog-wrapper"
+    actions={okButton}
+    open={open}
+    onRequestClose={() => setDialogVisibility(false)}
+  >
+    {message}
+    <p key='enroll'>The discount will be automatically applied when you enroll.</p>
```

Comment:
Can you remove `key` here? It's only necessary if you have an array of react components or a very complicated nested structure